caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java

```diff
@@ -0,0 +1,337 @@
+                        elasticAuditConfiguration.setClusterName(ES_CLUSTERNAME);
+                        return (T) elasticAuditConfiguration;
+                    }
+                });
+                com.hpe.caf.auditing.AuditChannel auditChannel = auditConnection.createChannel()
+        ) {
+
+            //  Index a sample audit event message into Elasticsearch.
+            AuditEventBuilder auditEventBuilder = auditChannel.createEventBuilder();
+
+            //  Set up fixed field data for the sample audit event message.
+            auditEventBuilder.setApplication(APPLICATION_ID);
+            auditEventBuilder.setEventType(EVENT_CATEGORY_ID, EVENT_TYPE_ID);
+            auditEventBuilder.setCorrelationId(CORRELATION_ID);
+            auditEventBuilder.setTenant(TENANT_ID);
+            auditEventBuilder.setUser(USER_ID);
+
+            //  Set up random test values for custom field data.
+            Random rand = new Random();
+
+            String docStringParamValue = "testStringParam";
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_STRING_PARAM_FIELD,CUSTOM_DOC_STRING_PARAM_FIELD, docStringParamValue);
+            int docIntParamValue = rand.nextInt();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_INT_PARAM_FIELD,CUSTOM_DOC_INT_PARAM_FIELD, docIntParamValue);
+            short docShortParamValue = (short) rand.nextInt(Short.MAX_VALUE + 1);
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_SHORT_PARAM_FIELD,CUSTOM_DOC_SHORT_PARAM_FIELD,docShortParamValue);
+            long docLongParamValue = rand.nextLong();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_LONG_PARAM_FIELD,CUSTOM_DOC_LONG_PARAM_FIELD, docLongParamValue);
+            float docFloatParamValue = rand.nextFloat();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_FLOAT_PARAM_FIELD,CUSTOM_DOC_FLOAT_PARAM_FIELD, docFloatParamValue);
+            double docDoubleParamValue = rand.nextDouble();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_DOUBLE_PARAM_FIELD,CUSTOM_DOC_DOUBLE_PARAM_FIELD, docDoubleParamValue);
+            boolean docBooleanParamValue = rand.nextBoolean();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_BOOLEAN_PARAM_FIELD,CUSTOM_DOC_BOOLEAN_PARAM_FIELD, docBooleanParamValue);
+            Date docDateParamValue = new Date();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_DATE_PARAM_FIELD,CUSTOM_DOC_DATE_PARAM_FIELD, docDateParamValue);
+
+            //  Send audit event message to Elasticsearch.
+            auditEventBuilder.send();
+            Thread.sleep(1000);
+
+            //  Search Elasticsearch for the newly indexed document (by UserId).
+            try ( TransportClient transportClient = ElasticAuditTransportClientFactory.getTransportClient(esHostNames, ES_PORT, ES_CLUSTERNAME)) {
+                SearchHit[] hits = searchDocument(transportClient, ES_INDEX, ES_TYPE, USER_ID_FIELD, USER_ID);
```

Comment:
You can keep the sleep if you put it in a loop with a overall limit to how many times you try to find the result. This way on fast machines the test still runs quickly, slower machines try multiple times before failing the test.

Also as we discussed with Chris, put something unique in the test ausit event so you can reliably find it and not find events from other tests by mistake.